pkg/kubernetes/storageclass/v1alpha1/storageclass.go

```diff
@@ -0,0 +1,110 @@
+package v1alpha1
+
+import (
+	storagev1 "k8s.io/api/storage/v1"
+)
+
+// StorageClass is a wrapper over API based
+// storage class instance
+type StorageClass struct {
+	object *storagev1.StorageClass
+}
+
+// StorageClassList holds the list of StorageClass instances
+type StorageClassList struct {
+	items []*StorageClass
+}
+
+// Predicate defines an abstraction
+// to determine conditional checks
+// against the provided StorageClass instance
+type Predicate func(*StorageClass) bool
+
+// predicateList holds the list of predicates
+type predicateList []Predicate
+
+// ListBuilder enables building an instance of StorageClassList
+type ListBuilder struct {
+	list    *StorageClassList
+	filters predicateList
+}
+
+// WithAPIList builds the list of StorageClass
+// instances based on the provided
+// StorageClass list
+func (b *ListBuilder) WithAPIList(scl *storagev1.StorageClassList) *ListBuilder {
+	if scl == nil {
+		return b
+	}
+	b.WithAPIObject(scl.Items...)
+	return b
+}
+
+// WithObject builds the list of StorageClass instances based on the provided
+// StorageClass list instance
+func (b *ListBuilder) WithObject(scs ...*StorageClass) *ListBuilder {
+	b.list.items = append(b.list.items, scs...)
+	return b
+}
+
+// WithAPIObject builds the list of node instances based on StorageClass api instances
+func (b *ListBuilder) WithAPIObject(scs ...storagev1.StorageClass) *ListBuilder {
+	for _, sc := range scs {
+		sc := sc // Pin it
+		b.list.items = append(b.list.items, &StorageClass{&sc})
+	}
+	return b
+}
+
```

Comment:
This function should not be used in this way. Rather move this logic to  `WithAPIList` function.

Reply:
The function should look like
```
func(b *Builder)WithAPIObject(sc *storagev1.StorageClass) *Builder{
...
}
```
Where `Builder` will encapsulate the custom StorageClass object. As we know, this StorageClass object is our custom object that in turn encapsulate the actual k8s apiserver storageclass object.